File: config/app-config.ts
// アプリ設定ファイル
// このファイルで3つのアプリのテキストと色を切り替えます

export type AppType = "END" | "SLEEP" | "DECIDE";

export interface AppConfig {
  appName: string;
  launchText: string;
  executionTexts: string[];
  endText: string;
  dotColor: string; // アイコンのドット色
  bundleId: {
    ios: string;
    android: string;
  };
}

export const APP_CONFIGS: Record<AppType, AppConfig> = {
  END: {
    appName: "END 3min",
    launchText: "終業前の3分を開始",
    executionTexts: [
      "残っている感覚が薄れていく",
      "抜ける方向に\nゆっくり吐く",
      "思考は遠ざかる\n考えは使われない",
      "さっきより弱い\nそれで十分",
    ],
    endText: "今日は切り替わった気がする",
    dotColor: "#FFFFFF", // 白
    bundleId: {
      ios: "com.timer.end3min",
      android: "com.timer.end3min",
    },
  },
  SLEEP: {
    appName: "SLEEP 3min",
    launchText: "寝る前の3分を開始",
    executionTexts: [
      "残っている感覚を\n一つだけ感じる",
      "吐く終わりで\n抜ける方向に",
      "言葉が出たら\n「今は使わない」と一回だけ",
      "さっきより弱い、が一つあれば十分",
    ],
    endText: "このまま何もしないでください",
    dotColor: "#0000FF", // 青
    bundleId: {
      ios: "com.timer.sleep3min",
      android: "com.timer.sleep3min",
    },
  },
  DECIDE: {
    appName: "NOW NOT DECIDE",
    launchText: "いま決めたい感覚を\n一つだけ感じる",
    executionTexts: [
      "決めたい感覚が\n薄れていく",
      "抜ける方向に\nゆっくり吐く",
      "言葉が出たら\n「今は決めない」と一回だけ",
      "「さっきより弱い」が一つあれば終了",
    ],
    endText: "終わりました\n何もしないでください",
    dotColor: "#FF0000", // 赤
    bundleId: {
      ios: "com.timer.nownotdecide",
      android: "com.timer.nownotdecide",
    },
  },
};

// 環境変数またはビルド時に設定されるアプリタイプ
// デフォルトは END（開発時）
export const getAppType = (): AppType => {
  try {
    // Expoのextra設定から取得
    const Constants = require("expo-constants").default;
    const appType = Constants?.expoConfig?.extra?.appType as
      | AppType
      | undefined;
    if (appType && ["END", "SLEEP", "DECIDE"].includes(appType)) {
      return appType;
    }
  } catch (e) {
    // expo-constantsが利用できない場合
  }

  // 環境変数から取得（ビルド時に設定）
  const envAppType = process.env.EXPO_PUBLIC_APP_TYPE as AppType | undefined;
  if (envAppType && ["END", "SLEEP", "DECIDE"].includes(envAppType)) {
    return envAppType;
  }

  // デフォルト
  return "END";
};

export const getAppConfig = (): AppConfig => {
  return APP_CONFIGS[getAppType()];
};

File: flutter_app/lib/config/app_config.dart
// App configuration for the three variants: END, SLEEP, DECIDE.

enum AppType { end, sleep, decide }

class AppConfig {
  final String appName;
  final String launchText;
  final List<String> executionTexts;
  final String endText;
  final String dotColor;
  final String iosBundleId;
  final String androidPackage;

  const AppConfig({
    required this.appName,
    required this.launchText,
    required this.executionTexts,
    required this.endText,
    required this.dotColor,
    required this.iosBundleId,
    required this.androidPackage,
  });
}

const Map<AppType, AppConfig> appConfigs = {
  AppType.end: AppConfig(
    appName: 'END 3min',
    launchText: '終業前の3分を開始',
    executionTexts: [
      '残っている感覚が薄れていく',
      '抜ける方向に\nゆっくり吐く',
      '思考は遠ざかる\n考えは使われない',
      'さっきより弱い\nそれで十分',
    ],
    endText: '今日は切り替わった',
    dotColor: '#FFFFFF',
    iosBundleId: 'com.timer.end3min',
    androidPackage: 'com.timer.end3min',
  ),
  AppType.sleep: AppConfig(
    appName: 'SLEEP 3min',
    launchText: '寝る前の3分を開始',
    executionTexts: [
      '残っている感覚',
      '抜ける方向に\nゆっくり吐く',
      '言葉が出ても\n使わない',
      '弱くなっている\nそれで十分',
    ],
    endText: 'おやすみなさい',
    dotColor: '#0000FF',
    iosBundleId: 'com.timer.sleep3min',
    androidPackage: 'com.timer.sleep3min',
  ),
  AppType.decide: AppConfig(
    appName: 'NOW NOT DECIDE',
    launchText: 'いま決めたい感覚を\n一つだけ感じる',
    executionTexts: [
      '決めたい感覚が\n薄れていく',
      '抜ける方向に\nゆっくり吐く',
      '言葉は浮かばない\n今は決めない',
      '弱くなっている\nそれで十分',
    ],
    endText: '何もしない',
    dotColor: '#FF0000',
    iosBundleId: 'com.timer.nownotdecide',
    androidPackage: 'com.timer.nownotdecide',
  ),
};

// Default app type (e.g. for single build). Use --dart-define or build flavors to override.
AppType getAppType() {
  const env = String.fromEnvironment(
    'APP_TYPE',
    defaultValue: 'decide',
  );
  switch (env.toUpperCase()) {
    case 'END':
      return AppType.end;
    case 'SLEEP':
      return AppType.sleep;
    case 'DECIDE':
    default:
      return AppType.decide;
  }
}

AppConfig getAppConfig() => appConfigs[getAppType()]!;
